go/private/cgo.bzl

```diff
@@ -161,47 +160,25 @@ _cgo_filter_srcs = rule(
     implementation = _cgo_filter_srcs_impl,
     attrs = {
         "srcs": attr.label_list(),
-        "strip_prefix": attr.string(),
-        "out_dir": attr.string(mandatory = True),
+        "id": attr.string(mandatory = True),
         #TODO(toolchains): Remove _toolchain attribute when real toolchains arrive
         "_go_toolchain": attr.label(default = Label("@io_bazel_rules_go_toolchain//:go_toolchain")),
     },
     fragments = ["cpp"],
-    output_to_genfiles = True,
 )
 
 def _cgo_select_files_impl(ctx):
-  out_dir = ctx.attr.out_dir
-
   outputs = []
-  cmds = []
   for src in ctx.files.srcs:
     match = any([src.path.endswith(ext) for ext in ctx.attr.exts])
     if ctx.attr.invert:
       match = not match
     if match:
-      dst = ctx.new_file(out_dir + "/" + _strip_prefix(src.path, ctx.attr.strip_prefix))
-      outputs += [dst]
-      cmds += [
-          "mkdir -p '%s'" % dst.dirname,
-          "cp '%s' '%s'" % (src.path, dst.path)
-      ]
-  if ctx.outputs.out_empty_hack:
-    outputs += [ctx.outputs.out_empty_hack]
-    cmds += [
-        "mkdir -p '%s'" % ctx.outputs.out_empty_hack.dirname,
-        "echo -n >'%s'" % ctx.outputs.out_empty_hack.path,
-    ]
-    
-  if outputs:
-    f = emit_generate_params_action(cmds, ctx, ctx.label.name + ".params")
-    ctx.action(
-        inputs = [f] + ctx.files.srcs,
-        outputs = outputs,
-        mnemonic = "CgoSelectFiles",
-        command = f.path,
-    )
-
+      outputs += [src]
+  if ctx.attr.at_least_one_file and not outputs:
+    empty_file = ctx.new_file("empty_file" + ctx.attr.exts[0])
```

Comment:
This is all about to change again as I merge it down and replace it with go code...